integration-tests/bats/fetch.bats

```diff
@@ -0,0 +1,342 @@
+    cd $TMPDIRS/repo1
+    dolt sql -q "create table t1 (a int primary key, b int)"
+    dolt add .
+    dolt commit -am "First commit"
+    dolt sql -q "insert into t1 values (0,0)"
+    dolt commit -am "Second commit"
+    dolt branch feature
+    dolt push origin main
+    cd $TMPDIRS
+}
+
+teardown() {
+    teardown_common
+    rm -rf $TMPDIRS
+    cd $BATS_TMPDIR
+}
+
+@test "fetch: basic fetch" {
+    cd repo2
+    dolt fetch
+
+    run dolt diff main origin/main
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "added table" ]] || false
+
+    run dolt sql -q "show tables as of hashof('origin/main')" -r csv
+    [ "${#lines[@]}" -eq 2 ]
+    [[ "$output" =~ "Table" ]] || false
+    [[ "$output" =~ "t1" ]] || false
+}
+
+@test "fetch: fetch origin" {
+    cd repo2
+    dolt fetch origin
+
+    run dolt diff main origin/main
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "added table" ]] || false
+
+    run dolt sql -q "show tables as of hashof('origin/main')" -r csv
+    [ "${#lines[@]}" -eq 2 ]
+    [[ "$output" =~ "Table" ]] || false
+    [[ "$output" =~ "t1" ]] || false
+}
+
+@test "fetch: fetch main" {
+    cd repo1
+    dolt checkout feature
+    dolt sql -q "create table t2 (a int primary key, b int)"
+    dolt add .
+    dolt commit -am "Third commit"
+    dolt push origin feature
+    cd ..
+
+    cd repo2
+    dolt fetch origin main
+
+    run dolt diff main origin/main
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "added table" ]] || false
+
+    run dolt sql -q "show tables as of hashof('origin/main')" -r csv
+    [ "${#lines[@]}" -eq 2 ]
+    [[ "$output" =~ "Table" ]] || false
+    [[ "$output" =~ "t1" ]] || false
+
+    run dolt branch -r
+    [ "$status" -eq 0 ]
+    [[ ! "$output" =~ "feature" ]] || false
+}
+
+@test "fetch: fetch custom remote" {
+    cd repo2
+    dolt fetch test-remote
+
+   run dolt diff main test-remote/main
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "added table" ]] || false
+
+    run dolt sql -q "show tables as of hashof('test-remote/main')" -r csv
+    [ "${#lines[@]}" -eq 2 ]
+    [[ "$output" =~ "Table" ]] || false
+    [[ "$output" =~ "t1" ]] || false
+}
+
+@test "fetch: fetch specific ref" {
+    cd repo2
+    dolt fetch test-remote refs/heads/main:refs/remotes/test-remote/main
+
+    run dolt diff main test-remote/main
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "added table" ]] || false
+
+    run dolt sql -q "show tables as of hashof('test-remote/main')" -r csv
+    [ "${#lines[@]}" -eq 2 ]
+    [[ "$output" =~ "Table" ]] || false
+    [[ "$output" =~ "t1" ]] || false
+}
+
+@test "fetch: fetch feature branch" {
+    cd repo1
+    dolt push origin feature
+
+    cd ../repo2
+    dolt fetch origin feature
+
+    run dolt diff main origin/feature
+    [ "$status" -eq 0 ]
+    [[ "$output" =~ "added table" ]] || false
+
+    run dolt sql -q "show tables as of hashof('origin/feature')" -r csv
+    [ "${#lines[@]}" -eq 2 ]
+    [[ "$output" =~ "Table" ]] || false
```

Comment:
Same comment

It's not enough to just push / fetch different refs, the assertions need to rule out that you fetched a different one than you intended

Reply:
Same applies to all the tests below. Put a new commit / table on the thing you're fetching so you know you got the right thing